src/methods.rs

```diff
@@ -92,6 +98,59 @@ impl LateLintPass for MethodsPass {
                     }
                 }
             }
+            // check Option.map(_).unwrap_or(_)
+            else if name.node.as_str() == "unwrap_or" {
+                if let ExprMethodCall(ref inner_name, _, ref inner_args) = args[0].node {
+                    if inner_name.node.as_str() == "map"
+                            && match_type(cx, cx.tcx.expr_ty(&inner_args[0]), &OPTION_PATH) {
+                        //get args to map() and unwrap_or()
+                        let map_arg = snippet(cx, inner_args[1].span, "..");
+                        let unwrap_arg = snippet(cx, args[1].span, "..");
+                        //if either is > 1 line, replace with 'f' and 'g'
+                        let (map_snippet, unwrap_snippet) =
+                            if map_arg.lines().count() > 1 || unwrap_arg.lines().count() > 1 {
+                                ("f".into(), "g".into())
+                            }
+                            else {
+                                (map_arg, unwrap_arg)
+                            };
+                        //lint
+                        span_lint(
+                            cx, OPTION_MAP_UNWRAP_OR, expr.span,
+                            &format!("called `map({0}).unwrap_or({1})` on an Option value. This \
```

Comment:
We avoid changing the lint message itself. Keep the existing lint message with f and g

Instead, use `span_lint_and_note` in case the thing one line, or just `span_lint` if it isn't.